### PR TITLE
DAOS-2781 build: upgrade SPDK to v19.04.1

### PR DESCRIPTION
**Pull request by NiuYawei**
Upgrade SPDK to v19.04.1

Signed-off-by: Niu Yawei <yawei.niu@intel.com>
Change-Id: I37b9b42e34463b4297b0137ef2499bdd741d4a03

### CODE REVIEW

File: sl_patches/spdk_export_set_thread.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
+diff --git a/include/spdk/thread.h b/include/spdk/thread.h
```
Comment (brianjmurrell) [inline sl_patches/spdk_export_set_thread.patch]: So this is the patch that you will apply in the RPM build also then.

File: components/__init__.py [hunk 2]
```diff
@@ -342,13 +342,14 @@ def define_components(reqs):
...
                 retriever=retriever,
-                commands=['./configure --prefix="$SPDK_PREFIX" ' \
+                commands=['./configure --prefix="$SPDK_PREFIX" --with-shared ' \
                           ' --with-fio="$FIO_SRC"',
                           'make $JOBS_OPT', 'make install',
                           'cp dpdk/build/lib/* "$SPDK_PREFIX/lib"',
                           'mkdir -p "$SPDK_PREFIX/share/spdk"',
                           'cp -r include scripts examples/nvme/fio_plugin ' \
                           '"$SPDK_PREFIX/share/spdk"'],
+		patch='$PATCH_PREFIX/spdk_export_set_thread.patch',
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply components/__init__.py]: Should fix this one.  You can run ./check_modules.sh -c to check internal ones to see if you fix it locally

[the remark this answers is not in the document]
Comment (NiuYawei) [reply components/__init__.py]: yes, sure.

File: sl_patches/spdk_export_set_thread.patch (hunk 1 above)
Comment (NiuYawei) [reply sl_patches/spdk_export_set_thread.patch]: Yes, I'm not quite sure how to make this change in the daos spdk repository, could you help to submit a patch for this change? Thanks in advance.

**Comment by NiuYawei** [comment]
Please hold on landing this, I'm testing DAOS with SPDK v19.04.1 now, there are still some issues needs be fixed.

**Comment by NiuYawei** [comment]
> Please hold on landing this, I'm testing DAOS with SPDK v19.04.1 now, there are still some issues needs be fixed.

Well, that was a nasty blobstore compatibility issue. I've addressed it in the DAOS patch  and will refreshe the DAOS patch later (once got a question confirmed from SPDK team), so this patch could be landed if no objections.

**Pull request merged**